source/common/http/conn_pool_base.cc

```diff
@@ -1,16 +1,52 @@
 #include "common/http/conn_pool_base.h"
 
+#include "common/common/assert.h"
+#include "common/network/transport_socket_options_impl.h"
+#include "common/runtime/runtime_features.h"
 #include "common/stats/timespan_impl.h"
 #include "common/upstream/upstream_impl.h"
 
 namespace Envoy {
 namespace Http {
+Network::TransportSocketOptionsSharedPtr
+wrapTransportSocketOptions(Network::TransportSocketOptionsSharedPtr transport_socket_options,
+                           Protocol protocol) {
+  if (!Runtime::runtimeFeatureEnabled("envoy.reloadable_features.http_default_alpn")) {
+    return transport_socket_options;
+  }
+
+  // If configured to do so, we override the ALPN to use for the upstream connection to match the
+  // selected protocol.
+  std::string alpn;
+  switch (protocol) {
+  case Http::Protocol::Http10:
+    NOT_REACHED_GCOVR_EXCL_LINE;
+  case Http::Protocol::Http11:
+    alpn = "http/1.1";
```

Comment:
Do we store these ALPN constants anywhere else? Wondering if we should not repeat this in various places? I assume we at least use them during HCM downstream codec selection?

Reply:
Also does it make sense to stash this as a utility somewhere?  not sure if the mobile folks would reuse.